### ISSUE TITLE
Identify OS specific libraries + Build order + Correct output type

### ISSUE THREAD

**Issue by TamasBaloghDev** [pavelliavonau/cmakeconverter#141]
Pavel, fantastic tool! Saved me a ton of time! Thank you very much Sir! 

I am adding these smaller insights when **I was porting a Win application to Linux.** I am not asking to implement these, I fixed these myself manually because it was very little effort, I just hope one day you can polish this tool and implement some other fantastic ideas I saw among the issue thread to license this application to make some money out of this (Haha, maybe I am biased but I really like this tool and would be willing to pay a reasonable amount to use it ;-) ) ! 

1) Do not copy OS specific libraries to additional library dependencies
![image](https://user-images.githubusercontent.com/34321407/166107796-643fe33b-337a-4c14-b003-f1766af36f4b.png)

2) For multiple projects in a solution, identify correct build order. This is done by using add_subdirectory(..) in the root level CMakeLists file but the order is wrong (assuming dependency among the projects)

3) Identify correct asset type: executable, static library, shared library?
For me it would always want to create executables for the shared libraries..

Thanks again, good luck!
Tamas Balogh

**Comment by pavelliavonau** [comment]
Hi, Tamas,
Thank you for so bright and kind compliments. I appreciate it. It's open source, man, I won't going to change the license) But if you want make me happier - you may a do donation :)

Now with points:
1. It's not a bug. any libraries should be copied to CMake to make the same set on Windows. I met this case by myself and solved it with "#pragma lib" in a file for windows only. I may recommend the same solution for you.
2. add_subdirectory does not affect build order. Subdirectories are just sorted in alphabetical order.
3. Everything is working fine. I have no idea what are you talking about. I need a sample to check it.

**Comment by TamasBaloghDev** [comment]
Hi Pavel,
1. Awesome, that I did not know!
2. Yesss, I later realised this when I looked inside the project level CMake files!
3. I know why this happened to me, I only have this solution configured in Debug|x64 mode only (early stage project..), and only for one configuration the output type is checked, so it will add "add_executable" because it assumes the same output type throughout (which is ... a fair assumption). 

Thank you again, I have added a small donation! So, I will close this ticket now, all solved!
